src/gui/powermanagement/powermanagement.h

```diff
@@ -28,38 +29,38 @@
 
 #pragma once
 
-#include <QObject>
+#include <QtSystemDetection>
 
-#ifdef Q_OS_MACOS
-// Require Mac OS X >= 10.5
-#include <IOKit/pwr_mgt/IOPMLib.h>
+#if defined(QBT_USES_DBUS)
+#include "inhibitordbus.h"
+using InhibitorBase = InhibitorDBus;
+#elif defined(Q_OS_MACOS)
+#include "inhibitormacos.h"
+using InhibitorBase = InhibitorMacOS;
+#elif defined(Q_OS_WIN)
+#include "inhibitorwindows.h"
+using InhibitorBase = InhibitorWindows;
+#else
+#include "inhibitor.h"
+using InhibitorBase = Inhibitor
 #endif
 
-#ifdef QBT_USES_DBUS
-class PowerManagementInhibitor;
-#endif
-
-class PowerManagement final : public QObject
+class PowerManagement final : public InhibitorBase
```

Comment:
I dislike this change. I would keep Inhibitor instance aggregated instead (as it was before).

Reply:
Is it a strong opposition?
IMO separating the high-level logic (the PowerManagement class in this PR) from the various `Inhibitor` implementations looks better to me.

Reply:
Or did you mean having something like this `InhibitorBase *m_inhibitor = nullptr;` as a private variable?

Reply:
> Or did you mean having something like this `InhibitorBase *m_inhibitor = nullptr;` as a private variable instead of inheriting?

Yes.

Reply:
> ~If I'm not mistaken, this way it will have an additional pointer/redirection compared to class inheritance.~

Is that a problem? Certainly not in this case. In addition, it allows to move some implementation details out from the header.

Reply:
>Is that a problem? Certainly not in this case. In addition, it allows to move some implementation details out from the header.

It doesn't buy us many advantages. The implementation class is fixed at compile time. I'll make it a non-pointer for now.

Reply:
>It doesn't buy us many advantages. 

This buys us the same advantages as any other code structuring.
Does this part of the code really require some kind of micro optimization to neglect other things?
(In the general case, polymorphism would even allow you to choose a specific implementation at runtime, if such an opportunity were available, for example, if there are several available ways to use on some platform.)

BTW, instances of `QObject` are supposed to be used indirectly (not by value).

Reply:
>BTW, instances of QObject are supposed to be used indirectly (not by value).

This is a valid concern. Changed.
All done.